src/components/Navbar.jsx

```diff
@@ -0,0 +1,189 @@
+        padding: 15px;
+
+        @media (min-width: 761px){
+            display: none;
+        }
+    `,
+    NavbarExtendedContainer : styled.div`
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        height: ${(props) => (props.extendNavbar ? "325px" : "80px")};
+        width: 100vw;
+
+        @media (min-width: 761px){
+            display: none;
+        }
+    `,
+    NavbarExtendedLinkContainer: styled.div`
+        margin-top: 80px;
+        display: flex;
+        flex-direction: column;  
+        align-items: center;
+        background: orange;    
+        width: 100%;             
+    `,
+    NavbarLinkExtended: styled(Link)`
+        color: white;
+        font-size: 25px;
+        font-family: Arial, Helvetica, sans-serif;
+        text-decoration: none;
+        margin: 10px;
+    `
+}
+
+
+function Navbar() {
+    const [extendNavbar, setExtendNavbar] = useState(false);
+    const [cartButton, setCartButton] = useState(false);
+    console.log(extendNavbar);
+    console.log(cartButton);
+
+
+    return <S.NavbarContainer extendNavbar={extendNavbar}>
+        <S.NavbarInnerContainer>
+            <S.LeftContainer>
+                <S.OpenLinkButton onClick={() => {
+                    if(extendNavbar === false) {
+                        setCartButton(false);
+                    }
+                    setExtendNavbar(!extendNavbar);
+                }}>{extendNavbar ? <> &#10005;</> : <>&#8801;</>}</S.OpenLinkButton>
+                <S.NavbarLogo src={Logo}></S.NavbarLogo>
+                <S.NavbarLinkContainer>
+                    <S.NavbarLink to="/">Home</S.NavbarLink>
+                    <S.NavbarLink to="/products">Products</S.NavbarLink>
+                    <S.NavbarLink to="/about">About</S.NavbarLink>
+                    <S.NavbarLink to="/contact">Contact</S.NavbarLink>
+                </S.NavbarLinkContainer>                 
+            </S.LeftContainer>
+            <S.RightContainer>
+                <S.CartButton onClick={() => {
+                    if(cartButton === false) {
+                        setExtendNavbar(false);
+                    }
+                    setCartButton(!cartButton);
+                }}><S.CartLogo src={CartLogo}></S.CartLogo></S.CartButton>
+                <S.Avatar src={AvatarImage}></S.Avatar>
+            </S.RightContainer>
+        </S.NavbarInnerContainer>
+
+        { extendNavbar && (
+            <S.NavbarExtendedContainer>            
+                <S.NavbarExtendedLinkContainer>
+                    <S.NavbarLinkExtended to="/">Home</S.NavbarLinkExtended>
+                    <S.NavbarLinkExtended to="/products">Products</S.NavbarLinkExtended>
+                    <S.NavbarLinkExtended to="/about">About</S.NavbarLinkExtended>
+                    <S.NavbarLinkExtended to="/contact">Contact</S.NavbarLinkExtended>
+                </S.NavbarExtendedLinkContainer>   
+            </S.NavbarExtendedContainer>
```

Comment:
We can create a separate functional component for this section,
Doing so will keep this current component tidy and less nested